move-to-infra/move-to-infra.sh

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env bash
+
+# This script moves ARO apps into intrastructure nodes.
+# It follows RHEL guide: https://access.redhat.com/solutions/5034771
+#
+# Infra nodes are:
+#  label: node-role.kubernetes.io/infra: ""
+#  taints:
+#  - key: infra
+#    value: reserved
+#    effect: NoSchedule
+#  - key: infra
+#    value: reserved
+#    effect: NoExecute
+
+
+# Check if pods are running
+checkStatus()
+{
+  LABEL="$1"
+  NAMESPACE="$2"
+  EXPECTED_PODS="$3"
+
+  echo "...checking pod labels ${LABEL} at namespace ${NAMESPACE}..."
+  ATTEMPTS=0
+  until [[ $ATTEMPTS -ge 10 ]]
+  do
+    PODS_READY=$(oc get pod  -l "${1}" --field-selector=status.phase==Running -o jsonpath="{.items[*].metadata.name}" -n"${NAMESPACE}" | wc -w)
+    if [[ $PODS_READY -eq $EXPECTED_PODS ]]; then
+      echo "...found expected pods running"
+      break
+    fi
+    ATTEMPTS=$((ATTEMPTS+1))
+    echo "Sleep 10s and check again."
+    sleep 10
+  done
+  if [[ $PODS_READY -ne $EXPECTED_PODS ]]; then
+    echo "...not found expected pods. exit"
+    exit 1
+  fi
+}
+
+# create config map to move
+monitoringConfigMap()
+{
+  cat <<EOT | oc apply -f -
+apiVersion: v1
```

Comment:
I would like to resurrect this conversation instead of "silent resolving"
Please share your concerns/thoughts regarding below:

Having yaml file inline definitely has its own pros.
But the concern I have, is that YAML files can't be tested if they're defined inline. But extracting it into separate files you get the following benefits:

1. ability to lint yamls: https://github.com/pre-commit/pre-commit-hooks#check-yaml
1. increase readability of shell files
1. reduced cognitive load when updating/maintaining scripts 
1. scalability. At the moment it targets only monitoring applications, but in the future, we may want to expand the list.

Reply:
All these make a a lot of sense. As long as having separate yaml file will not cause any issues with where and how the script is run - I'm fine with it.

Reply:
ok, updated to have separate file